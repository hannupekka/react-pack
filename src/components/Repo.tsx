import * as React from 'react';
import { pure } from 'recompose';

interface Props {
  url: string;
  user: string;
  name: string;
}

const Repo = (props: Props) => (
  <a href={props.url} title={`author: ${props.user}`} target="_blank" rel="noopener noreferrer">
    {props.name}
  </a>
);

export default pure(Repo);
