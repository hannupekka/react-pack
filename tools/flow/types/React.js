declare type ElementType = React$Element<any>
