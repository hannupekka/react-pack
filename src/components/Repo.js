// @flow
import React from 'react';
import { pure } from 'recompose';
import type { Component } from 'recompose';

type Props = {
  name: string,
  url: string,
  user: string
}

const Repo: Component<Props> = (props: Props): React$Element<any> => {
  return (
    <a
      href={props.url}
      title={`author: ${props.user}`}
      target="_blank"
      rel="noopener noreferrer"
    >
      {props.name}
    </a>
  );
};

export default pure(Repo);
