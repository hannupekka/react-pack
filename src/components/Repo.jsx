import React from 'react';
import PropTypes from 'prop-types';
import { pure } from 'recompose';

const Repo = props => (
  <a href={props.url} title={`author: ${props.user}`} target="_blank" rel="noopener noreferrer">
    {props.name}
  </a>
);

Repo.propTypes = {
  url: PropTypes.string.isRequired,
  user: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
};

export default pure(Repo);
