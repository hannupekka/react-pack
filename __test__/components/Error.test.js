import React from 'react';
import renderer from 'react-test-renderer';
import Repo from 'components/Repo';

describe('Error', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <Repo
        name="repository name"
        url="https://github.com"
        user="repository owner"
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
