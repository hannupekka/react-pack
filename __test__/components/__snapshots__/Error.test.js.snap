// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error renders correctly 1`] = `
<a
  href="https://github.com"
  rel="noopener noreferrer"
  target="_blank"
  title="author: repository owner"
>
  repository name
</a>
`;
